File: leetcode/C++/compare-version-numbers.cpp
class Solution {
public:
    void parsingVersion(string &version, vector<int> &v){
        string s;
        for(char &ch : version){
            if(ch=='.'){
                if(s.length()>0) v.push_back(stoi(s));
                s.clear();
            }
            else s.push_back(ch);
        }
        if(s.length()>0) v.push_back(stoi(s));
    }
    int compareVersion(string version1, string version2) {
        vector<int> v1, v2;
        int idx_1=0, idx_2=0;
        parsingVersion(version1, v1); parsingVersion(version2, v2);
        for(; idx_1<v1.size()&& idx_2<v2.size(); ++idx_1, ++idx_2){
            if(v1[idx_1]>v2[idx_2]) return 1;
            else if(v1[idx_1]<v2[idx_2]) return -1;
        }
        for(; idx_1<v1.size(); ++idx_1) if(v1[idx_1]>0) return 1;
        for(; idx_2<v2.size(); ++idx_2) if(v2[idx_2]>0) return -1;
        return 0;
    }
};

File: leetcode/Python/compare-version-numbers.py
class Solution:
    def compareVersion(self, version1: str, version2: str) -> int:
        def versionParsing(version: str):
            v, s = list(), ''
            for ch in version:
                if ch == '.':
                    if len(s)>0:
                        v.append(int(s))
                    s = ''
                else:
                    s += ch
            if len(s)>0:
                v.append(int(s))
            return v
        v1, v2 = versionParsing(version1), versionParsing(version2)
        idx_1, idx_2 = 0, 0
        while idx_1<len(v1) and idx_2<len(v2):
            if v1[idx_1]>v2[idx_2]:
                return 1
            elif v1[idx_1]<v2[idx_2]:
                return -1
            idx_1, idx_2 = idx_1 + 1 , idx_2 + 1
        while idx_1<len(v1):
            if v1[idx_1] > 0:
                return 1
            idx_1 += 1
        while idx_2 < len(v2):
            if v2[idx_2] > 0:
                return -1
            idx_2 += 1
        return 0